crates/bevy_asset/src/asset_server.rs

```diff
@@ -125,18 +137,38 @@ impl AssetServer {
             .read()
             .get(extension)
             .map(|index| self.server.loaders.read()[*index].clone())
-            .ok_or_else(|| AssetServerError::MissingAssetLoader(Some(extension.to_string())))
+            .ok_or_else(|| AssetServerError::MissingAssetLoader {
+                extensions: vec![extension.to_string()],
+            })
     }
 
     fn get_path_asset_loader<P: AsRef<Path>>(
         &self,
         path: P,
     ) -> Result<Arc<Box<dyn AssetLoader>>, AssetServerError> {
-        path.as_ref()
-            .extension()
-            .and_then(|e| e.to_str())
-            .ok_or(AssetServerError::MissingAssetLoader(None))
-            .and_then(|extension| self.get_asset_loader(extension))
+        let s = path
+            .as_ref()
+            .file_name()
+            .ok_or(AssetServerError::MissingAssetLoader {
+                extensions: Vec::new(),
+            })?
+            .to_str()
+            .ok_or(AssetServerError::MissingAssetLoader {
+                extensions: Vec::new(),
+            })?;
+
+        let mut exts = Vec::new();
+        let mut ext = s;
+        while let Some(idx) = ext.find('.') {
+            ext = &ext[idx + 1..];
+            exts.push(ext);
+            if let Ok(loader) = self.get_asset_loader(ext) {
+                return Ok(loader);
+            }
+        }
+        Err(AssetServerError::MissingAssetLoader {
+            extensions: exts.drain(0..).map(String::from).collect(),
```

Comment:
O think you can use `into_iter()` instead of `drain(0..)`.

Reply:
instead of `0..`, I'd reccommend `..`

Reply:
Ah yeah, I was trying to figure that out. I forgot a bout `into_iter`. :+1: